src/php/Command/FormatCommand.php

```diff
@@ -4,23 +4,75 @@
 
 namespace Phel\Command;
 
+use Phel\Command\Shared\CommandIoInterface;
 use Phel\Formatter\FormatterInterface;
 
 final class FormatCommand
 {
     public const COMMAND_NAME = 'fmt';
 
+    private const PHEL_EXTENSION = 'phel';
+
     private FormatterInterface $formatter;
+    private CommandIoInterface $io;
 
-    public function __construct(FormatterInterface $formatter)
-    {
+    /** @var list<string> */
+    private array $filePaths = [];
+
+    public function __construct(
+        FormatterInterface $formatter,
+        CommandIoInterface $io
+    ) {
         $this->formatter = $formatter;
+        $this->io = $io;
+    }
+
+    public function run(array $paths): void
+    {
+        foreach ($paths as $fileOrPath) {
+            $this->runFileOrPath($fileOrPath);
+        }
+
+        $this->printResult();
+    }
+
+    private function runFileOrPath(string $fileOrPath): void
+    {
+        if (is_dir($fileOrPath)) {
+            $this->runFormatterInDirectory($fileOrPath);
+        } elseif ($this->hasPhelExtension($fileOrPath)) {
+            $wasFormatted = $this->formatter->formatFile($fileOrPath);
+
+            if ($wasFormatted) {
+                $this->filePaths[] = $fileOrPath;
+            }
+        }
+    }
+
+    private function runFormatterInDirectory(string $directory): void
+    {
+        $paths = array_diff(scandir($directory), ['..', '.']);
```

Comment:
This is not looking for files in subfolders. I would recommend this approach:
```
$iterator = new RecursiveIteratorIterator(
  new RecursiveDirectoryIterator($directory),
  RecursiveIteratorIterator::LEAVES_ONLY
);
```

Reply:
Thanks, I updated the code to use your suggestion.